rllib/utils/replay_buffers/episode_replay_buffer.py

```diff
@@ -0,0 +1,508 @@
+                            idx_cursor = i + 1
+                            break
+                if idx_cursor is not None:
+                    new_indices.extend(self._indices[idx_cursor:])
+                self._indices = new_indices
+
+                # Increase episode ejected counter.
+                self._num_episodes_ejected += 1
+
+    @override(ReplayBufferInterface)
+    def sample(
+        self,
+        num_items: Optional[int] = None,
+        *,
+        batch_size_B: Optional[int] = None,
+        batch_length_T: Optional[int] = None,
+    ) -> SampleBatchType:
+        """Returns a batch of size B (number of "rows"), where each row has length T.
+
+        Each row contains consecutive timesteps from an episode, but might not start
+        at the beginning of that episode. Should an episode end within such a
+        trajectory, a random next episode (starting from its t0) will be concatenated
+        to that "row". For more details, see the docstring of the EpisodeReplayBuffer
+        class.
+
+        Args:
+            num_items: See `batch_size_B`. For compatibility with the
+                `ReplayBufferInterface` abstract base class.
+            batch_size_B: The number of rows (trajectories) to return in the batch.
+            batch_length_T: The length of each row (in timesteps) to return in the
+                batch.
+
+        Returns:
+            The sampled batch (observations, actions, rewards, terminateds, truncateds)
+                of dimensions [B, T, ...].
+        """
+        if num_items is not None:
+            assert batch_size_B is None, (
+                "Cannot call `sample()` with both `num_items` and `batch_size_B` "
+                "provided! Use either one."
+            )
+            batch_size_B = num_items
+        # Use our default values if no sizes/lengths provided.
+        batch_size_B = batch_size_B or self.batch_size_B
+        batch_size_T = batch_size_T or self.batch_size_T
+
+        # Uniformly sample n samples from self._indices (all stores timesteps).
+        index_tuples_idx = []
+        observations = [[] for _ in range(batch_size_B)]
+        actions = [[] for _ in range(batch_size_B)]
+        rewards = [[] for _ in range(batch_size_B)]
+        is_first = [[False] * batch_length_T for _ in range(batch_size_B)]
+        is_last = [[False] * batch_length_T for _ in range(batch_size_B)]
+        is_terminated = [[False] * batch_length_T for _ in range(batch_size_B)]
+        is_truncated = [[False] * batch_length_T for _ in range(batch_size_B)]
+
+        B = 0
+        T = 0
+        idx_cursor = 0
+        while B < batch_size_B:
+            # Ran out of uniform samples -> Sample new set.
+            if len(index_tuples_idx) <= idx_cursor:
+                index_tuples_idx.extend(list(np.random.randint(
+                    0, len(self._indices), size=batch_size_B * 10
+                )))
```

Comment:
Can you describe the reasoning behind the size of the sampled indices being `batch_size_B * 10`? Why don't we add for example 1?

Reply:
I think you are right. Doing it this way is pretty haywire. I'll fix it and always simply sample a single value. That should be cleaner and easier to understand.

Reply:
done